agent/consul/leader_connect.go

```diff
@@ -94,16 +94,61 @@ func parseCARoot(pemValue, provider, clusterID string) (*structs.CARoot, error)
 	}, nil
 }
 
+// parseCAIntermediate returns a filled-in structs.CAIntermediate from a raw PEM value.
+func parseCAIntermediate(pemValue, provider, clusterID string) (*structs.CAIntermediate, error) {
+	id, err := connect.CalculateCertFingerprint(pemValue)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing intermediate fingerprint: %v", err)
+	}
+	cert, err := connect.ParseCert(pemValue)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing intermediate cert: %v", err)
+	}
+	return &structs.CAIntermediate{
+		ID:                  id,
+		Name:                fmt.Sprintf("%s CA Intermediate Cert", strings.Title(provider)),
+		SerialNumber:        cert.SerialNumber.Uint64(),
+		SigningKeyID:        connect.HexString(cert.SubjectKeyId),
+		ExternalTrustDomain: clusterID,
+		NotBefore:           cert.NotBefore,
+		NotAfter:            cert.NotAfter,
+		IntermediateCert:    pemValue,
+		Active:              true,
+	}, nil
+}
```

Comment:
See other comment but the fact this is almost 100% copy paste of `parseCARoot` doesn't feel right.

Reply:
You're right. Originally I thought I was going to need more than just the signing key.